include/itkTorchTensorAccessorHelper.h

```diff
@@ -0,0 +1,273 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkTorchTensorAccessorHelper_h
+#define itkTorchTensorAccessorHelper_h
+
+#include <torch/torch.h>
+#include "itkImageBase.h"
+#include "itkTorchPixelHelper.h"
+
+namespace itk
+{
+/** \class TorchTensorAccessorHelper
+ *  \brief Efficiently uses underlying torch::TensorAccessor object.
+ *
+ * Access to pixel information via an `IndexType index` is achieved
+ * with an underlying torch::TensorAccessor via multiple applications
+ * of operator[].  First the last component of `index` is used, which
+ * is the slowest varying component in the underlying pixel memory.
+ * Then the second-to-last component, etc.  This class automates that
+ * repeated application of the operator[].
+
+ * This is the non-specialization, which can be used for an
+ * arbitrarily large number of dimensions in `IndexType index`.
+ *
+ * \ingroup PyTorch
+ */
+template< typename DeepScalarType, typename IndexType, int VCurrentAccessorLevel, int VNumberOfSteps >
+class TorchTensorAccessorHelper
+{
+public:
+  static torch::TensorAccessor< DeepScalarType, VCurrentAccessorLevel-VNumberOfSteps > &
+  FindPixel( torch::TensorAccessor< DeepScalarType, VCurrentAccessorLevel > &accessor, const IndexType &index )
+    {
+    // The `index` variable has the dimension that varies fastest in
+    // the underlying buffer at index[0], but `accessor` uses
+    // indices starting with the slowest varying dimension first.
+    return TorchTensorAccessorHelper< DeepScalarType, IndexType, VCurrentAccessorLevel-1, VNumberOfSteps-1 >::FindPixel( accessor[index[VNumberOfSteps-1]], index );
+    }
+};
+
+/** \class TorchTensorAccessorHelper
+ *  \brief Efficiently uses underlying torch::TensorAccessor object.
+ *
+ * Access to pixel information via an `IndexType index` is achieved
+ * with an underlying torch::TensorAccessor via multiple applications
+ * of operator[].  First the last component of `index` is used, which
+ * is the slowest varying component in the underlying pixel memory.
+ * Then the second-to-last component, etc.  This class automates that
+ * repeated application of the operator[].
+
+ * This is the specialization for which 0 additional applications of
+ * operator[] are required.
+ *
+ * \ingroup PyTorch
+ */
+template< typename DeepScalarType, typename IndexType, int VCurrentAccessorLevel >
+class TorchTensorAccessorHelper< DeepScalarType, IndexType, VCurrentAccessorLevel, 0 >
+{
+public:
+  static torch::TensorAccessor< DeepScalarType, VCurrentAccessorLevel > &
+  FindPixel( torch::TensorAccessor< DeepScalarType, VCurrentAccessorLevel > &accessor, const IndexType & /* index */ )
+    {
+    return accessor;
+    }
+};
+
+/** Specializations for VNumberOfSteps > 0 are not strictly necessary
+ * but are implemented for speed. */
```

Comment:
Is this an assumption? Or did you try it both ways? If there is no speed difference, less code is preferred.

Reply:
There are no image/region iterators for TorchImage (yet) so the speed of these accessors could be quite important.  I'll have to do some speed tests.